src/libmongoc/src/mongoc/mongoc-cmd.c

```diff
@@ -981,12 +981,24 @@ mongoc_cmd_parts_assemble (mongoc_cmd_parts_t *parts,
 
       if (!is_get_more) {
          if (cs) {
+            /* Snapshot Sessions Spec: "Snapshot reads require MongoDB 5.0+."
+             * Throw an error if snapshot is enabled and wire version is less
+             * than 13 before potentially appending "snapshot" read concern. */
+            if (mongoc_session_opts_get_snapshot (&cs->opts) &&
+                server_stream->sd->max_wire_version <
+                   WIRE_VERSION_SNAPSHOT_READS) {
+               bson_set_error (error,
+                               MONGOC_ERROR_TRANSACTION,
+                               MONGOC_ERROR_TRANSACTION_INVALID_STATE,
```

Comment:
Is there a better domain and/or code for this error?

Reply:
I think the `MONGOC_ERROR_TRANSACTION` domain should be avoided. This error scenario does not necessarily mean a transaction was started on this session.

The [definition](http://mongoc.org/libmongoc/current/errors.html) of the error code `MONGOC_ERROR_CLIENT_SESSION_FAILURE` seems fitting:
> Failure related to creating or using a logical session.

It is currently used in client side checks related to sessions:
- validating causal consistency is not enabled with snapshot
- checking the server indicates session support
- failure to generate a UUID locally

I suggest using `MONGOC_ERROR_CLIENT_SESSION_FAILURE` with the domain of `MONGOC_ERROR_CLIENT`.

Reply:
That sounds perfect; done.